src/cloudcasting/dataset.py

```diff
@@ -183,6 +192,90 @@ def __getitem__(self, key: DataIndex) -> tuple[NDArray[np.float32], NDArray[np.f
 
 
+class ValidationSatelliteDataset(SatelliteDataset):
+    def __init__(
+        self,
+        zarr_path: list[str] | str,
+        history_mins: int,
+        forecast_mins: int = 180,
+        sample_freq_mins: int = 15,
+        nan_to_num: bool = False,
+    ):
+        """A torch Dataset used only in the validation proceedure.
+
+        Args:
+            zarr_path: Path to the satellite data for validation. Can be a string or list
+            history_mins: How many minutes of history will be used as input features
+            forecast_mins: How many minutes of future will be used as target features
+            sample_freq_mins: The sample frequency to use for the satellite data
+            nan_to_num: Whether to convert NaNs to -1.
+        """
+
+        super().__init__(
+            zarr_path=zarr_path,
+            start_time=None,
+            end_time=None,
+            history_mins=history_mins,
+            forecast_mins=forecast_mins,
+            sample_freq_mins=sample_freq_mins,
+            preshuffle=True,
+            nan_to_num=nan_to_num,
+        )
+
+    @staticmethod
+    def _find_t0_times(
+        date_range: pd.DatetimeIndex, history_mins: int, forecast_mins: int, sample_freq_mins: int
+    ) -> pd.DatetimeIndex:
+        # Find the valid t0 times for the available data. This avoids trying to take samples where
+        # there would be a missing timestamp in the sat data required for the sample
+        available_t0_times = find_valid_t0_times(
+            date_range, history_mins, forecast_mins, sample_freq_mins
+        )
+
+        # Get the required 2022 test dataset t0 times
+        val_t0_times_from_csv = get_required_test2022_t0_times()
+
+        # Find the intersection of the available t0 times and the required validation t0 times
+        val_time_available = val_t0_times_from_csv.isin(available_t0_times)
+
+        # Make sure all of the required validation times are available in the data
+        if not val_time_available.all():
+            msg = (
+                "The following validation t0 times are not available in the satellite data: \n"
+                f"{val_t0_times_from_csv[~val_time_available]}\n"
+                "The validation proceedure requires these t0 times to be available."
+            )
+            raise ValueError(msg)
+
+        return val_t0_times_from_csv
+
+
+def _get_t0_times(path: str) -> pd.DatetimeIndex:
+    """Get the required validation t0 times"""
+
+    # Load the zipped csv file as a byte stream
+    data = pkgutil.get_data("cloudcasting", path)
+    if data is not None:
+        byte_stream = io.BytesIO(data)
+    else:
+        # Handle the case where data is None
+        msg = f"No data found for path: {path}"
+        raise ValueError(msg)
+
+    # Load the times into pandas
+    df = pd.read_csv(byte_stream, encoding="utf8", compression="zip")
+
+    return pd.DatetimeIndex(df.t0_time)
+
+
+def get_required_test2022_t0_times() -> pd.DatetimeIndex:
```

Comment:
Why is `_get_t0_times` underscore prefixed (soft-marked as private) but the more specific `get_required_test2022_t0_times` is not? Shouldn't this be the other way around (or make both the same)?